src/network/networkd-ndisc.c

```diff
@@ -451,13 +451,18 @@ static int ndisc_router_process_onlink_prefix(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return log_link_warning_errno(link, r, "Failed to get prefix length: %m");
 
+        r = sd_ndisc_router_get_preference(rt, &preference);
```

Comment:
Could you comment that Prefix Information option does not have preference, hence we use the 'main' preference here ?